src/App/Fossa/Configuration.hs

```diff
@@ -0,0 +1,92 @@
+{-# LANGUAGE TypeApplications #-}
+{-# LANGUAGE TemplateHaskell #-}
+
+module App.Fossa.Configuration
+( mergeFileCmdMetadata
+, readConfigFile
+, ConfigFile (..)
+, ConfigProject (..)
+, ConfigRevision (..)
+)
+where
+
+import Data.Aeson ( FromJSON(parseJSON), (.:), (.:?), withObject )
+import Data.Text (Text)
+import App.Types
+import Path
+import qualified Control.Carrier.Diagnostics as Diag
+import Effect.ReadFS
+import Options.Applicative
+
+data ConfigFile = ConfigFile
+  { version :: Int
+  , server :: Maybe Text
+  , apiKey :: Maybe Text
+  , project :: Maybe ConfigProject
+  , revision :: Maybe ConfigRevision
+  } deriving (Eq, Ord, Show)
+
+data ConfigProject = ConfigProject
+  { projID :: Maybe Text
+  , name :: Maybe Text
+  , link :: Maybe Text
+  , team :: Maybe Text
+  , jiraKey :: Maybe Text
+  , url :: Maybe Text
+  , policy :: Maybe Text
+  } deriving (Eq, Ord, Show)
+
+data ConfigRevision = ConfigRevision
+  { commit :: Maybe Text
+  , branch :: Maybe Text
+  } deriving (Eq, Ord, Show)
+
+instance FromJSON ConfigFile where
+  parseJSON = withObject "ConfigFile" $ \obj ->
+    ConfigFile <$> obj .: "version"
+               <*> obj .:? "server"
+               <*> obj .:? "apiKey"
+               <*> obj .:? "project"
+               <*> obj .:? "revision"
+
+instance FromJSON ConfigProject where
+  parseJSON = withObject "ConfigProject" $ \obj ->
+    ConfigProject <$> obj .:? "id"
+                  <*> obj .:? "name"
+                  <*> obj .:? "link"
+                  <*> obj .:? "team"
+                  <*> obj .:? "jirakey"
+                  <*> obj .:? "url"
+                  <*> obj .:? "policy"
+
+instance FromJSON ConfigRevision where
+  parseJSON = withObject "ConfigRevision" $ \obj ->
+    ConfigRevision <$> obj .:? "commit"
+                   <*> obj .:? "branch"
+
+defaultFile :: Path Rel File
+defaultFile = $(mkRelFile ".fossa.yml")
+
+readConfigFile :: IO ConfigFile
+readConfigFile = do
+      file <- Diag.runDiagnosticsIO $ runReadFSIO $ readContentsYaml @ConfigFile defaultFile
+      case file of
+        Left _ -> pure $ ConfigFile { version = -1
+                                    , server = Nothing
+                                    , apiKey = Nothing
+                                    , project = Nothing
+                                    , revision = Nothing
+                                    }
+        Right a -> pure $ Diag.resultValue a 
```

Comment:
If we encounter **any** error, we're just ignoring config altogether, which I strongly disagree with.  My reasoning is that a user may expect their config to supply some info, and we simply ignore it silently and do something entirely different.

We should at least check for the file missing and then ignore, and if parsing went wrong we should crash with a good message.

Reply:
Yea I agree with you, I think crashing if the file exists and we fail is correct,